catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/dialog/utils/LegoSensorPortConfigDialogWrapper.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.ui.dialog.utils;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.hamcrest.core.StringStartsWith.startsWith;
+
+public final class LegoSensorPortConfigDialogWrapper{
+
+	private static final int[] FORMULA_EDITOR_LEGO_PORTS = {R.string.lego_port_1, R.string.lego_port_2,
+			R.string.lego_port_3, R.string.lego_port_4 };
+
+	private LegoSensorPortConfigDialogWrapper() {
+		String legoSensorConfigDialogTitle = UiTestUtils.getResourcesString(R.string.lego_sensor_port_config_dialog_title);
+		legoSensorConfigDialogTitle = legoSensorConfigDialogTitle.replace("%1$s!", "");
```

Comment:
theres android resources string methods that build the strings with numbers in them for you, please use those. have a look at:
https://developer.android.com/guide/topics/resources/string-resource#formatting-strings

Reply:
i know, i had it in the first place. Its not a number, but the name of the sensor wich is selected.
So the wrapper would have to be called with the resString of the Sensor.

Reply:
in the test you know which sensor youre expecting there, then also assert for this.